src/buffer.rs

```diff
@@ -19,12 +19,24 @@ pub struct Cursor {
     pub line: usize,
     /// First-byte-index of glyph at cursor (will insert behind this glyph)
     pub index: usize,
+    /// Whether to associate the cursor with the run before it (false) or the run after it (true)
+    /// if placed at the boundary between two runs
+    pub affinity: bool,
```

Comment:
I feel like it would be clearer if this was an enum with `Before` and `After` variants instead of a boolean.